### ISSUE TITLE
Variable with name "end"

### ISSUE THREAD

**Issue by LukasWallisch** [jqlang/jq#1619]
Hey guy's i'm not sure if this was already reported and i overlooked it.
If a predefined variable is given the name "end", jq throws an error: "syntax error, unexpected end..."
for example:
`$ jq -n --arg last "1" '{end: $last}'`
works as expected giving:
`{
  "end": "1"
}` 

but
`$ jq -n --arg end "1" '{end: $end}'`
gives:
`jq: error: syntax error, unexpected end, expecting IDENT or __loc__ (Unix shell quoting issues?) at <top-level>, line 1:
{end: $end}       
jq: 1 compile error`

I don't know how complex it would be to fix this, but at least it should be mentioned in the manual **I** think.

**Comment by nicowilliams** [comment]
I'm trying to find a way to permit `$<keyword>`, but it causes tons of reduce-reduce conflicts from Bison...

**Comment by pkoppstein** [comment]
@LukasWallisch - The manual probably omits the restriction because it was not intended to be definitive.  In the meantime, the jq [FAQ ](https://github.com/stedolan/jq/wiki/FAQ) mentions the issue in the `Caveats` section.

**Comment by nicowilliams** [comment]
I think we'll have to change the lexer to produce a separate token type for `$name` and the grammar accordingly.

**Comment by Alanscut** [comment]
hi @LukasWallisch  
Now jq can handle the scenario you described ，you can try it again.
jq version : master branch

```
root@os-001: jq -n --arg end "1" '{end: $end}'
{
  "end": "1"
}


```

**Comment by itchyny** [comment]
Same issue of #1868 and fixed by c72ed135e4f1b5a02a8fb3f6cd46f27513ecab2a, so closing the issue.
